src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SeparateSidewalkTagCheck.java

```diff
@@ -0,0 +1,309 @@
+    {
+        final PolyLine tLine = new PolyLine(location);
+        return line.segments().stream()
+                .min(Comparator.comparingDouble(s -> s.shortestDistanceTo(tLine).asMeters()))
+                .orElseThrow(() -> new CoreException("Unable to get a Segment {}", line));
+    }
+
+    /**
+     * Generate and return flag for separately mapped sidewalk if tags are not consistent with the
+     * highway’s sidewalk tags.
+     *
+     * @param edge
+     *            edge to flag
+     * @param sidewalkTagValue
+     *            separate sidewalk tag value
+     * @return an optional {@link CheckFlag} object
+     */
+    private Optional<CheckFlag> generateFlag(final Edge edge, final String sidewalkTagValue)
+    {
+        super.markAsFlagged(edge.getOsmIdentifier());
+        return Optional.of(this.createFlag(new OsmWayWalker(edge).collectEdges(),
+                this.getLocalizedInstruction(0, edge.getOsmIdentifier(), sidewalkTagValue)));
+    }
+
+    /**
+     * Return sidewalk tag value. Also mapping alternative sidewalk tagging to preferred.
+     * https://wiki.openstreetmap.org/wiki/Key:sidewalk
+     *
+     * @param edge
+     *            edge to get sidewalk tag values.
+     * @return separate sidewalk tag value.
+     */
+    @SuppressWarnings("squid:S3655")
+    private String getSeparateSidewalkTagValue(final Edge edge)
+    {
+        String sidewalkTagValue = null;
+
+        // preferred way of mapping sidewalks
+        if (edge.getTag(SidewalkTag.KEY).isPresent())
+        {
+            sidewalkTagValue = edge.getTag(SidewalkTag.KEY).get();
+        }
+
+        // alternative way of mapping sidewalk
+        if (edge.getTag(SidewalkLeftTag.KEY).isPresent()
+                && separatelyMappedTagValue.equals(edge.getTag(SidewalkLeftTag.KEY).get())
+                && (edge.getTag(SidewalkRightTag.KEY).isPresent()
+                        && "no".equals(edge.getTag(SidewalkRightTag.KEY).get())
+                        || edge.getTag(SidewalkRightTag.KEY).isEmpty()))
+        {
+            sidewalkTagValue = "left";
+        }
+
+        if (edge.getTag(SidewalkRightTag.KEY).isPresent()
+                && separatelyMappedTagValue.equals(edge.getTag(SidewalkRightTag.KEY).get())
+                && (edge.getTag(SidewalkLeftTag.KEY).isPresent()
+                        && "no".equals(edge.getTag(SidewalkLeftTag.KEY).get())
+                        || edge.getTag(SidewalkLeftTag.KEY).isEmpty()))
+        {
+            sidewalkTagValue = "right";
+        }
+
+        return sidewalkTagValue;
+    }
+
+    /**
+     * Verifies that separate sidewalk and highway are within acceptable heading degree range
+     *
+     * @param headingOne
+     *            {@link Heading} one to check
+     * @param headingTwo
+     *            {@link Heading} to check.
+     * @return {@code true} if both heading are within acceptable heading degree range
```

Comment:
maybe explain why this is used? I don't remember, but I remember you told me and it made sense.